Assets/Plugins/LeapMotion/Core/Editor/LeapServiceProviderEditor.cs

```diff
@@ -77,63 +97,641 @@ public class LeapServiceProviderEditor : CustomEditorBase<LeapServiceProvider> {
     }
 
     public virtual void OnSceneGUI() {
-      Vector3 origin = target.transform.TransformPoint(controllerOffset);
 
-      Vector3 local_top_left, top_left, local_top_right, top_right,
-              local_bottom_left, bottom_left, local_bottom_right, bottom_right;
-      getLocalGlobalPoint(-1, 1, 1, out local_top_left, out top_left);
-      getLocalGlobalPoint(1, 1, 1, out local_top_right, out top_right);
-      getLocalGlobalPoint(-1, 1, -1, out local_bottom_left, out bottom_left);
-      getLocalGlobalPoint(1, 1, -1, out local_bottom_right, out bottom_right);
+      switch (GetSelectedInteractionVolume()) {
+        case LeapServiceProvider.InteractionVolumeVisualization.None:
+          break;
+        case LeapServiceProvider.InteractionVolumeVisualization.LeapMotionController:
+          DrawLeapMotionControllerInteractionZone(LMC_BOX_WIDTH, LMC_BOX_DEPTH, LMC_BOX_RADIUS, Color.white);
+          break;
+        case LeapServiceProvider.InteractionVolumeVisualization.RigelPlaceholderNotAccurate:
+          DrawRigelInteractionZoneMesh();
+          break;
+        case LeapServiceProvider.InteractionVolumeVisualization.Automatic:
+          DetectConnectedDevice();
+          break;
+        default:
+          break;
+      }
+
+    }
+
+    private void ParseRigelInteractionMeshData() {
+
+      if (rigelInteractionZoneMesh == null) {
+        try
+        {
+          ObjFileParser rigelMeshDataParser = new ObjFileParser();
+
+          // This data is not oriented or scaled correctly yet.
+          rigelInteractionZoneMesh = rigelMeshDataParser.FromObj(Path.Combine(Application.dataPath, "UnityModules", "Assets", "Plugins", "LeapMotion", "Core", "Models", "Rigel-interaction-cone-placeholder.obj"),
```

Comment:
Loading an .obj via full path parsing is obscenely brittle; so much so that it simply doesn't work on my machine.
The result of that `Path.Combine()` on my end is:
`C:/Repositories/UnityModules/Assets\UnityModules\Assets\Plugins\LeapMotion\Core\Models\Rigel-interaction-cone-placeholder.obj`
While mixed forward/backward slash paths can be fixed with `myPath = Path.GetFullPath(myPath);`, the correct way to do all this is to use the fact that Unity can parse/import .obj models directly.

Replace it and all the OBJ Parsing code with something like:
[Mesh rigelInteractionZoneMesh = (Mesh)AssetDatabase.LoadAssetAtPath(Path.Combine("Assets", "Plugins", "LeapMotion", "Core", "Models", "Rigel-interaction-cone-placeholder.obj"), typeof(Mesh));](https://docs.unity3d.com/ScriptReference/AssetDatabase.LoadAssetAtPath.html)

And then use the fact that Unity pre-parses it to just iterate over all the triangle connections to get the "edges".